identity/client/src/main/java/com/azure/identity/credential/MSICredential.java

```diff
@@ -0,0 +1,37 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.identity.credential;
+
+import com.azure.identity.credential.msi.MSIResourceType;
+import com.azure.identity.implementation.MSIToken;
+import com.azure.identity.implementation.RefreshableTokenCredential;
+
+public abstract class MSICredential extends RefreshableTokenCredential<MSIToken> {
+    MSICredential() {
+        super();
+    }
+
+    public static AppServiceMSICredential appService() {
```

Comment:
@JonathanGiles I don't like this design, open to your opinions. What I'm trying to achieve is that users don't have to know about different implementations but can create them starting from MSICredential. I don't want to merge the 2 implementations in one file cuz there will likely be more azure resources supporting MSI.

Reply:
For now I would remove these convenience methods and just expect users to use the AppServiceMSICredential, etc directly.